server/router.c

```diff
@@ -74,12 +80,36 @@ int routeTraffic(int connectionfd) {
         }
     }
 
-    // Send message
-    printf("[router] Responding to connection %d with %s regarding route %s\n", connectionfd, getVolatilePrintableResponseType(sendBuff), getVolatilePrintableResponseType(recvBuff));
-    int sendSuccess = sendMessage(connectionfd, sendBuff, SEND_BUFF_SIZE, MSG_DONTWAIT);
-    if (!sendSuccess) {
-        success = 0;
-        fprintf(stderr, "[router] Sending response failed for connection %d\n", connectionfd);
+    if (globalFlag) {
+        // Send response to all joined players
+        int gameID;
+
+        if (isMessageType(recvBuff, PROTOCOL_START_GAME_TYPE)) {
+            gameID = ((ProtocolStartGameRequest *) recvBuff)->gameID;
+        } else if (isMessageType(recvBuff, PROTOCOL_JOIN_GAME_TYPE)) {
+            gameID = ((ProtocolJoinGameRequest *) recvBuff)->gameID;
+        }
+
+        for (int i = 0; i < MAX_GAME_PLAYERS; i++) {
+            if (games[gameID].player[i].created) {
+                connectionfd = games[gameID].player[i].connectionfd;
+
+                int sendSuccess = sendMessage(connectionfd, sendBuff, SEND_BUFF_SIZE, MSG_DONTWAIT);
```

Comment:
In order to stay consistent, you could printf a debug message here just like in non-global mode.